app/src/main/java/com/henagle/www/kanban/DeckFragment.java

```diff
@@ -0,0 +1,106 @@
+package com.henagle.www.kanban;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ListView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.henagle.www.kanban.model.Card;
+import com.henagle.www.kanban.model.Deck;
+import com.henagle.www.kanban.model.Page;
+
+/**
+ * Created by rebeccahe on 5/20/18.
+ *
+ * Instances of this class are fragments representing a single card list.
+ */
+
+public class DeckFragment extends Fragment {
+
+    private ListView deckListView;
+
+    private Deck deck;
+    private DeckAdapter deckAdapter;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate layout for this fragment
+        View rootView = inflater.inflate(R.layout.fragment_deck, container, false);
+
+        // Retrieve deck data for this fragment
+        Bundle args = getArguments();
+        Page page = Page.values()[args.getInt(DeckPagerAdapter.CURRENT_PAGE_ARG)];
+        deck = MainActivity.decks.get(page);
+
+        // Update list view with deck data
+        deckListView = rootView.findViewById(R.id.deck_list_view);
+        deckAdapter = new DeckAdapter(getContext(), deck);
+        deckListView.setAdapter(deckAdapter);
+
+        return rootView;
+    }
+
+    /**
+     * This is to handle the delete button being clicked on a Card
+     * @param view view of Card to be deleted
+     */
+    protected void deleteClickHandler(View view) {
+        RelativeLayout viewParentRow = (RelativeLayout) view.getParent();
+        int index = ((ViewGroup) viewParentRow.getParent()).indexOfChild(viewParentRow);
+        if (deck.size() > index) {
+            deck.remove(index);
+        }
+        deckAdapter.notifyDataSetChanged();
+    }
```

Comment:
I moved the delete functionality into the Fragment, since that is what will be interacting with the Card data.